sparkle/src/components/SearchInput.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { forwardRef } from "react";
+
+import { Button, Icon, Input } from "@sparkle/components";
+import { MagnifyingGlassIcon, XMarkIcon } from "@sparkle/icons";
+import { cn } from "@sparkle/lib/utils";
+
+export interface SearchInputProps {
+  placeholder?: string;
+  value: string | null;
+  onChange?: (value: string) => void;
+  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+  name: string;
+  disabled?: boolean;
+  className?: string;
+}
+
+export const SearchInput = forwardRef<HTMLInputElement, SearchInputProps>(
+  (
+    {
+      placeholder = "Search",
+      value,
+      onChange,
+      onKeyDown,
+      name,
+      disabled = false,
+      className,
+    },
+    ref
+  ) => {
+    const clearInputField = () => {
+      onChange?.("");
+    };
+
+    return (
+      <div className={cn("s-relative s-m-px s-flex-grow", className)}>
+        <Input
+          type="text"
+          name={name}
+          placeholder={placeholder}
+          value={value}
+          onChange={(e) => {
+            onChange?.(e.target.value);
+          }}
+          onKeyDown={onKeyDown}
+          disabled={disabled}
+          ref={ref}
+        />
+        <div className="s-absolute s-inset-y-0 s-right-0 s-flex s-items-center s-pr-1">
+          {value && onChange ? (
+            <Button
+              icon={XMarkIcon}
+              variant="ghost"
+              size="xs"
+              onClick={clearInputField}
+            />
+          ) : (
+            <div
+              className={cn(
+                "s-px-2",
+                disabled ? "s-text-element-600" : "s-text-element-900"
+              )}
+            >
+              <Icon
+                visual={MagnifyingGlassIcon}
+                size="xs"
```

Comment:
does this make the icons smaller everywhere except in the `AssistantPicker`?

Reply:
Yup, they were a tiny bit big for non essential icons imo

Reply:
Agreed!